dockstore-yaml-proposals/xena/.dockstore.yml

```diff
@@ -17,32 +18,60 @@ service:
      - stop.sh
 
   scripts:
-    # The next 5 scripts are always executed in the order listed here.
-    presprovision: # Before any provisioning
-    prestart: # After provisioning, before starting the service
-    start: stand_up.sh # Starts the service
-    poststart: # After the service has started
-    postprovision: load.sh # After the service has started. For multiple provisionings, e.g., the next day
+    # The scripts preprovision, prestart, start, poststart, and postprovision will be executed in order.
+    # The only script required to have a value is start.
+    
+    # Before any provisioning
+    preprovision: 
+    
+    # After provisioning, before starting the service
+    prestart: 
+     
+    # Starts the service
+    start: bash stand_up.sh 
+    
+    # After the service has started
+    poststart: 
+    
+    # After the service has started. For multiple provisionings, e.g., the user may decide to add more data a week later.
+    postprovision: bash load.sh 
 
-    port: port.sh # Returns the port the service is exposing.
-    healthcheck: # Returns
-    stop: stop.sh # Stops the service
+    # Returns the port the service is exposing. Can be used by a client to know what port to proxy
+    port: bash port.sh
+    
+    # Returns an exit code of 0 if service is running successfully.
+    healthcheck: 
+    
+    # Stops the service
+    stop: bash stop.sh 
 
-  # Parameters are passed as environment variables to all of the above scripts. As such, their names need to be valid environment variable names.
-  # The exception is parameters of type dataset and dataset[] -- those parameters are provisioned.
+  # Parameters are passed to the scripts as environment variables. They are service
+  # specific, i.e., the service creator decides which values to make parameterizable.
   parameters:
-    httpPort:
+    # The name of the parameter, can be anything that is a valid environment variable name
+    httpPort: 
+        # An optional default value
         default: "7222"
-        description: The host's HTTP port.
-    httpsPort:
-        default: "7223"
-        description: The host's HTTPS port
-    dataset:
-        type: dataset[]
+        # A description that tools can use for prompting the a user.
+        description: The host's HTTP port. The default is 7222.
+
+  # Data are files that are provisioned locally for the service. Each key is a dataset name,
+  # where the dataset can contain 1 to n files
+  data:
+    dataset_1:
+        targetDirectory: xena/files # Can override this at the individual file level.
+
+        # The optional property indicates whether there can be multiple sets of files.
+        # If not present, or false, then only one set of files can be specified.
+        # In this case, Xena can accept multiple pairs of files, so it is true.
+        multiple: true
```

Comment:
Is this necessary? 
Devil's advocate, what if we define `files` to always hold an array?

Reply:
Attention to @garyluu who had an idea for further simplification for files based on CWL

Reply:
I've wondered about that too. There are two reasons why I've been hesitating:

1. Maybe some services don't accept multiple sets of data?
2. It makes it slightly more complicated if there aren't multiples, forcing the user to specify a one-element array, rather just one element.

That said, this does seem pretty clunky, so pending Gary's input, I'll go ahead and remove it.

Reply:
Removed `multiple` property in a commit soon to be pushed.